projects/manage-supervision-and-delius/src/main/kotlin/uk/gov/justice/digital/hmpps/service/SentenceService.kt

```diff
@@ -62,20 +63,19 @@ class SentenceService(
                 additionalSentences.map { it.toAdditionalSentence() }
             ),
             order = disposal?.toOrder(),
-            requirements = requirementRepository.getRequirements(id, eventNumber).map { it.toRequirement() },
-            courtDocuments = documentRepository.getCourtDocuments(id, eventNumber).map { it.toCourtDocument() }
+            requirements = requirementRepository.getRequirements(id, eventNumber)
+                .map { it.toRequirement(id, eventNumber) },
+            courtDocuments = documentRepository.getCourtDocuments(id, eventNumber).map { it.toCourtDocument() },
+            disposal?.id?.let { getUnpaidWorkTime(it) }
         )
 
     fun ExtraSentence.toAdditionalSentence(): AdditionalSentence =
         AdditionalSentence(length, amount, notes, type.description)
 
-    fun Person.toName() =
-        Name(forename, secondName, surname)
-
     fun Disposal.toOrder() =
         Order(description = type.description, length = length, startDate = date, endDate = expectedEndDate())
 
-    fun RequirementDetails.toRequirement(): Requirement {
+    fun RequirementDetails.toRequirement(eventId: Long, eventNumber: String): Requirement {
```

Comment:
Do you still need to pass the eventId and eventNumber?

Reply:
updated and fields are no longer used